src/transformers/pipelines/token_classification.py

```diff
@@ -210,63 +244,124 @@ def __call__(self, inputs: Union[str, List[str]], **kwargs):
-            "offset_mapping": offset_mapping,
-            "sentence": sentence,
-            **model_inputs,
-        }
+        return model_outputs
 
-    def postprocess(self, model_outputs, aggregation_strategy=AggregationStrategy.NONE, ignore_labels=None):
+    def postprocess(
+        self,
+        all_outputs,
+        aggregation_strategy=AggregationStrategy.NONE,
+        ignore_labels=None,
+        **postprocess_params,
+    ):
         if ignore_labels is None:
             ignore_labels = ["O"]
-        logits = model_outputs["logits"][0].numpy()
-        sentence = model_outputs["sentence"]
-        input_ids = model_outputs["input_ids"][0]
-        offset_mapping = model_outputs["offset_mapping"][0] if model_outputs["offset_mapping"] is not None else None
-        special_tokens_mask = model_outputs["special_tokens_mask"][0].numpy()
-
+        stride = postprocess_params.pop("stride", 0)
+        sentence = all_outputs[0]["sentence"]
+        keys = ["input_ids", "logits", "special_tokens_mask"]
+        # Check whether offset_mapping come from tokenizer or manual input
+        if "offset_mapping" in all_outputs[0].keys():
+            x = all_outputs[0]["offset_mapping"]
+            if self.framework == "tf":
+                if tf.is_tensor(x):
+                    keys.append("offset_mapping")
+            else:
+                if torch.is_tensor(x):
+                    keys.append("offset_mapping")
+        outputs = {}
+        for k in keys:
+            outputs[k] = np.array([model_outputs[k][0].numpy() for model_outputs in all_outputs])
+        logits = outputs.pop("logits")
         maxes = np.max(logits, axis=-1, keepdims=True)
         shifted_exp = np.exp(logits - maxes)
         scores = shifted_exp / shifted_exp.sum(axis=-1, keepdims=True)
 
-        if self.framework == "tf":
-            input_ids = input_ids.numpy()
-            offset_mapping = offset_mapping.numpy() if offset_mapping is not None else None
-
+        special_tokens_mask = outputs["special_tokens_mask"]
+
+        # Update each token scores with the maximum scores in all overlapping parts
+        for i in range(len(scores) - 1):
+            # Get current and next chunks without special tokens
+            current_mask = special_tokens_mask[i]
+            next_mask = special_tokens_mask[i + 1]
+
+            current_scores = scores[i][current_mask == 0]
+            next_scores = scores[i + 1][next_mask == 0]
+
+            chunk_length = len(current_scores)
+            # Stack the ovelapping part between the current and the next chunk
+            overlapping_scores = np.stack((current_scores[chunk_length - stride :], next_scores[:stride]))
+            # Get the highest score for each token in each chunk
+            tokens_max_scores = np.max(overlapping_scores, axis=-1)
+            # Get the chunk id of the highest score for each token
+            chunk_idx = np.argmax(tokens_max_scores, axis=0)
```

Comment:
"Here you are mix and matching hopeful that a more "confident" a token is ,the more likely it is but that's simply not true, adversarial attacks show that plenty." => Of course, but we have to choose one or maybe get entities independently for each chunk and merge results after this step. Again we'll have to choose between entities with different scores.

In fact even with the current implementation, if you provide a text with New York Bagels at the end with Bagels truncated, the model will be wrong due to the lack of context:
![image](https://user-images.githubusercontent.com/74506016/221588205-37176941-0513-4f7e-a69a-04a52a715f7e.png)


There is no perfect method of course. Keep the highest score is an acceptable solution I think or maybe keep the highest scores for the aggregated entities at the final step. I don't think we can do a lot more for that, we are restricted with the model's architecture. 

Maybe we can create a validation dataset to validate the approach? It's a nice feature for companies dealing with long texts.

Reply:
> There is no perfect method of course.

Entirely correct, which is why it cannot be on by default (we can't hide those tricks for users).

> I don't think we can do a lot more for that, 


I just proposed one above

> A very simple merge logic, could be to simply split the stride in half, use the first logits for the left part, and use the second logits for the next one. But this is still keeping the issue in the middle of the stride, but we can hope that in that case since both parts see the 2 tokens in a majority of time both chunks will agree. That needs to be heavily checked though.

I am not sure which is better, which is why we need some benchmarks real testing to gauge pros&cons. 
Like testing a few different (ideally different architectures) on a few different datasets. 
If you don't have the ressources to run complete benchmarks, you could stop at creating the scripts with extremely tiny amount of data and we could run it on larger data. 

I think here we need to be very pragmatic. I have concerns about your proposal from my theoretical understanding of models, but if it gets better results on representative tests, I'd be more than happy to be proven wrong.

My proposed approach seems naïve but is also relatively simple which is a huge strength in your proposal. 
I think aiming for simple and orthogonal to aggregation_strategy is really nice !

Reply:
I'm also experimenting a new approach:
- pass each chunk to aggregate function (without striding them) and get entities
- filter entities on the start index and keep the longest
I'll let you know

Reply:
Yes. I think any solution to this problem really requires the test I'm mentioning.

It's a bit too hard in my experience to anticipate how a strategy will preform.
Several architecture avoids overfitting a particular model, several datasets avoids overfitting English and a particular dataset.

I don't see any way around it.